MaterialLib/FractureModels/MohrCoulomb.h

```diff
@@ -0,0 +1,85 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MATERIALLIB_FRACTURE_MOHRCOULOMB_H_
+#define MATERIALLIB_FRACTURE_MOHRCOULOMB_H_
+
+#include <Eigen/Eigen>
+
+#include "ProcessLib/Parameter/Parameter.h"
+
+#include "FractureModelBase.h"
+
+namespace MaterialLib
+{
+namespace Fracture
+{
+
+template <int DisplacementDim>
+class MohrCoulomb final : public FractureModelBase<DisplacementDim>
+{
+public:
+    /// Variables specific to the material model
+    struct MaterialProperties
+    {
+        using P = ProcessLib::Parameter<double>;
+        using X = ProcessLib::SpatialPosition;
+
+        MaterialProperties(
+                P const& normal_stiffness, P const& shear_stiffness,
+                P const& friction_angle, P const& dilatancy_angle,
+                P const& cohesion)
+            : normal_stiffness(normal_stiffness), shear_stiffness(shear_stiffness),
```

Comment:
Small improvement:
A warning is issued "declaration shadows a field of MaterialLib::Fracture::MohrCoulomb::MaterialProperties" for all constructor's arguments. I usually postfix the arguments with underscore:
```c++
MaterialProperties(P const& normal_stiffness_) : normal_stiffness(normal_stiffness_) {}
```